File: app/models/construction_service.rb
# frozen_string_literal: true

class ConstructionService < ApplicationRecord
  validates :name, presence: true, uniqueness: true
  validates :unit, presence: true
  validates :price, presence: true, numericality: {greater_than_or_equal_to: 0}
  validates :category, presence: true
end

File: app/frontend/types/construction_service.ts
export interface ConstructionService {
  id: number;
  name: string;
  unit: string;
  price: number;
  category: string;
}

export interface PricingCalculatorIndexProps {
  serviceCategories: string[];
  servicesByCategory: Record<string, ConstructionService[]>;
  errors?: Record<string, string[]>;
}

export interface SelectedService {
  id: number;
  quantity: number;
}

export interface CalculatedPriceResponse {
  totalCost: number;
  errors?: Record<string, string[]>;
}
